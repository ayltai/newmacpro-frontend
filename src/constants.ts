import { Package, } from './types';

export const API_MAX_RETRIES           = 5;
export const HOMEBREW_REFRESH_INTERVAL = 1000 * 60 * 60 * 18;
export const TWEAKS_REFRESH_INTERVAL   = 1000 * 60 * 60 * 48;
export const BUNDLES_REFRESH_INTERVAL  = 1000 * 60 * 60 * 4;
export const MIN_SEARCH_LENGTH         = 2;
export const SEARCH_FUZZINESS          = 0.2;
export const SEARCH_MAX_RESULTS        = 50;

export const POPULAR_APPS : Package[] = [
    {
        id          : 'google-chrome',
        displayName : 'Google Chrome',
        description : 'Download the new Google Chrome for your iPhone and iPad. Now more simple, secure and faster than ever. Get the best of Google Search, and easily sync your bookmarks and passwords with Chrome on your laptop. Download the fast, secure browser recommended by Google.',
        source      : 'Homebrew/cask',
    },
    {
        id          : 'spotify',
        displayName : 'Spotify',
        description : 'With the Spotify music and podcast app, you can play millions of songs, albums and original podcasts for free. Stream music and podcasts, discover albums, playlists or even single songs for free on your mobile or tablet. Subscribe to Spotify Premium to download and listen offline wherever you are.',
        source      : 'Homebrew/cask',
    },
    {
        id          : 'alfred',
        displayName : 'Alfred',
        description : 'Alfred is an award-winning app for macOS which boosts your efficiency with hotkeys, keywords, text expansion and more. Search your Mac and the web, and be more productive with custom actions to control your Mac.',
        source      : 'Homebrew/cask',
    },
    {
        id          : 'visual-studio-code',
        displayName : 'Visual Studio Code',
        description : 'Visual Studio Code is a code editor redefined and optimized for building and debugging modern web and cloud applications.',
        source      : 'Homebrew/cask',
    },
    {
        id          : 'microsoft-office',
        displayName : 'Microsoft 365 for Mac',
        description : 'Office for Mac with Microsoft 365, gives you power and flexibility to get things done from virtually anywhere.',
        source      : 'Homebrew/cask',
    },
    {
        id          : 'notion',
        displayName : 'Notion',
        description : 'Notion is the connected workspace where better, faster work happens. Now with AI',
        source      : 'Homebrew/cask',
    },
    {
        id          : 'fantastical',
        displayName : 'Fantastical',
        description : 'Fantastical, the calendar app you won\'t be able to live without. Quickly create new events and reminders with natural language input and more.',
        source      : 'Homebrew/cask',
    },
    {
        id          : 'gimp',
        displayName : 'GIMP',
        description : 'GIMP is a cross-platform image editor available for GNU/Linux, macOS, Windows and more operating systems. It is free software, you can change its source code and distribute your changes.',
        source      : 'Homebrew/cask',
    },
    {
        id          : 'bartender',
        displayName : 'Bartender',
        description : 'Bartender is an award-winning app for macOS that superpowers your menu bar, giving you total control over your menu bar items, what\'s displayed, and when, with menu bar items only showing when you need them. Bartender improves your workflow with quick reveal, search, custom hotkeys and triggers, and lots more.',
        source      : 'Homebrew/cask',
    },
    {
        id          : '1password',
        displayName : '1Password',
        description : 'A password manager, digital vault, form filler and secure digital wallet. 1Password remembers all your passwords for you to help keep account information safe.',
        source      : 'Homebrew/cask',
    },
];

export const POPULAR_PACKAGES : Package[] = [
    {
        id          : 'node',
        displayName : 'Node.js',
        description : 'Node.jsÂ® is an open-source, cross-platform JavaScript runtime environment.',
        source      : 'Homebrew/core',
    },
    {
        id          : 'git',
        displayName : 'Git',
        description : 'Git is a free and open source distributed version control system designed to handle everything from small to very large projects with speed and efficiency. ',
        source      : 'Homebrew/core',
    },
    {
        id          : 'wget',
        displayName : 'Wget',
        description : 'GNU Wget is a free software package for retrieving files using HTTP, HTTPS, FTP and FTPS, the most widely used Internet protocols. It is a non-interactive commandline tool, so it may easily be called from scripts, cron jobs, terminals without X-Windows support, etc.',
        source      : 'Homebrew/core',
    },
    {
        id          : 'python3',
        displayName : 'Python 3',
        description : 'Python is a programming language that lets you work quickly and integrate systems more effectively.',
        source      : 'Homebrew/core',
    },
    {
        id          : 'awscli',
        displayName : 'AWS CLI',
        description : 'The AWS Command Line Interface (CLI) is a unified tool to manage your AWS services.',
        source      : 'Homebrew/core',
    },
    {
        id          : 'cmake',
        displayName : 'CMake',
        description : 'CMake is an open-source, cross-platform family of tools designed to build, test and package software.',
        source      : 'Homebrew/core',
    },
    {
        id          : 'maven',
        displayName : 'Maven',
        description : 'Maven is a build automation tool used primarily for Java projects.',
        source      : 'Homebrew/core',
    },
    {
        id          : 'jq',
        displayName : 'jq',
        description : 'jq is a lightweight and flexible command-line JSON processor.',
        source      : 'Homebrew/core',
    },
    {
        id          : 'docker',
        displayName : 'Docker',
        description : 'Docker is a set of platform as a service products that use OS-level virtualization to deliver software in packages called containers.',
        source      : 'Homebrew/core',
    },
    {
        id          : 'htop',
        displayName : 'htop',
        description : 'htop is an interactive process viewer for Unix systems.',
        source      : 'Homebrew/core',
    },
];

export const POPULAR_TWEAKS : Package[] = [
    {
        id          : 'tweak_macos_keep_windows',
        displayName : 'Always keep windows',
        description : 'macOS can remember all the tabs and windows you had open for the next time you launch it again.',
        source      : 'Tweak',
    },
    {
        id          : 'tweak_macos_show_hidden_files',
        displayName : 'Show hidden files in Finder',
        description : 'By default, macOS hides hidden files in Finder. This tweak disables this behavior.',
        source      : 'Tweak',
    },
    {
        id          : 'tweak_macos_disable_itunes_media_keys',
        displayName : 'Disable iTunes media keys',
        description : 'If you have iTunes installed, when you press Play on the keyboard, or connect the headset and Bluetooth speaker, iTunes will automatically run. This tweak disables this behavior.',
        source      : 'Tweak',
    },
    {
        id          : 'tweak_macos_subpixel_font_rendering',
        displayName : 'Change subpixel font rendering on external monitors',
        description : 'If you are running macOS on a Mac without a retina display, or with an external monitor that does not have an ultra-high resolution screen, you may have noticed that some fonts and text can appear as blurry and difficult to read. This is because macOS is rendering the text using subpixel font rendering.',
        source      : 'Tweak',
    },
    {
        id          : 'tweak_macos_disable_photos_auto_open',
        displayName : 'Disable Photos from opening automatically',
        description : 'macOS automatically opens Photos every time you insert a memory card or connect a device. This tweak disables this behavior.',
        source      : 'Tweak',
    },
    {
        id          : 'tweak_macos_show_file_extensions',
        displayName : 'Show file extensions in Finder',
        description : 'By default, macOS hides file extensions in Finder. This tweak disables this behavior.',
        source      : 'Tweak',
    },
    {
        id          : 'tweak_macos_disable_smart_quotes',
        displayName : 'Disable smart quotes',
        description : 'macOS automatically converts straight quotes to smart quotes. But for developers, it\'s annoying to have to manually change them back. This tweak disables this behavior.',
        source      : 'Tweak',
    },
    {
        id          : 'tweak_macos_disable_smart_dashes',
        displayName : 'Disable smart dashes',
        description : 'macOS automatically converts straight dashes to smart dashes. But for developers, it\'s annoying to have to manually change them back. This tweak disables this behavior.',
        source      : 'Tweak',
    },
];

type Document = {
    title   : string,
    docPath : string,
};

export const DOCUMENTS : Document[] = [
    {
        title   : 'footer.disclaimer',
        docPath : '/docs/disclaimer.md',
    },
    {
        title   : 'footer.privacy',
        docPath : '/docs/privacy.md',
    },
    {
        title   : 'footer.terms',
        docPath : '/docs/terms.md',
    },
];
