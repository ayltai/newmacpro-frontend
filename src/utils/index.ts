import * as Sentry from '@sentry/react';

export const handleError = (error : any) => {
    if (process.env.NODE_ENV === 'production') {
        Sentry.captureException(error);
    } else {
        console.error(error);
    }
};

export const capitalizeFirstLetter = (input : string) => input.charAt(0).toUpperCase() + input.slice(1);
