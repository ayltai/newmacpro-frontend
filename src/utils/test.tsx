import { ThemeProvider, } from '@mui/material';
import { configureStore, Store, } from '@reduxjs/toolkit';
import { act, render, } from '@testing-library/react';
import mediaQuery from 'css-mediaquery';
import { ConfirmProvider, } from 'material-ui-confirm';
import React, { ReactNode, } from 'react';
import { Provider, } from 'react-redux';

import { bundleApi, cartApi, } from '../apis';
import { POPULAR_APPS, } from '../constants';
import { cartReducer, prefReducer, store as reduxStore, } from '../states';
import { AppTheme, } from '../styles';
import { Cart, } from '../types';

export const mockApis = async (cart? : Cart) => {
    fetchMock.doMock(request => {
        if (request.url.endsWith('/api/bundles') && request.method === 'POST') return Promise.resolve({
            status : 201,
            body   : 'Created',
        });

        if (request.url.endsWith('/api/bundles/1') && request.method === 'GET') return Promise.resolve({
            status : 200,
            body   : JSON.stringify({
                id          : '1',
                displayName : 'Dummy',
                packages    : [
                    POPULAR_APPS[0],
                ],
            }),
        });

        if (request.url.endsWith('/api/bundles/1') && request.method === 'DELETE') return Promise.resolve({
            status : 200,
            body   : 'OK',
        });

        if (request.url.endsWith('/api/carts') && request.method === 'POST') return Promise.resolve({
            status : 201,
            body   : 'Created',
        });

        return Promise.resolve({
            status : 404,
            body   : 'Not found',
        });
    });

    await act(async () => {
        await reduxStore.dispatch(bundleApi.endpoints?.getBundle.initiate({
            bundleId : '1',
            idToken  : 'idToken',
        }));
        if (cart) await reduxStore.dispatch(cartApi.endpoints.addCart.initiate(cart));
    });
};

export const createMatchMedia = (width : number) => (query : string) => ({
    matches             : mediaQuery.match(query, {
        width,
    }),
    media               : '',
    onchange            : null,
    addListener         : () => {},
    removeListener      : () => {},
    addEventListener    : () => {},
    removeEventListener : () => {},
    dispatchEvent       : () => false,
});

export const createStore = (preloadedState? : Record<string, any>) => configureStore({
    preloadedState,
    reducer    : {
        cart                      : cartReducer,
        pref                      : prefReducer,
        [ bundleApi.reducerPath ] : bundleApi.reducer,
        [ cartApi.reducerPath   ] : cartApi.reducer,
    },
    middleware : getDefaultMiddleware => getDefaultMiddleware().concat(bundleApi.middleware, cartApi.middleware),
});

export const defaultStore = createStore();

export type RootState = ReturnType<typeof defaultStore.getState>;

const customRender = (ui : any, {
    preloadedState,
    store = createStore(preloadedState),
    ...rest
} : {
    preloadedState?   : Record<string, any>,
    store?            : Store<RootState>,
    [ rest : string ] : any,
} = {}) => render(ui, {
    wrapper : ({
        children,
    } : {
        children : ReactNode,
    }) => (
        <Provider store={store}>
            <ThemeProvider theme={AppTheme}>
                <ConfirmProvider>
                    {children}
                </ConfirmProvider>
            </ThemeProvider>
        </Provider>
    ),
    ...rest,
});

export * from '@testing-library/react';

export { customRender as render, };
