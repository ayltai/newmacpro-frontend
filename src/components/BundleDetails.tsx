import { Divider, Stack, TextField, Typography, } from '@mui/material';
import Filter from 'bad-words';
import React, { ChangeEvent, useEffect, } from 'react';
import { useTranslation, } from 'react-i18next';
import { adjectives, animals, colors, names, uniqueNamesGenerator, } from 'unique-names-generator';

import { Bundle, } from '../types';
import { capitalizeFirstLetter, } from '../utils';
import { ListView, } from './ListView';

export const BundleDetails = ({
    readOnly = true,
    bundle,
    onChange,
} : {
    readOnly? : boolean,
    bundle    : Bundle,
    onChange? : (bundle : Bundle) => void,
}) => {
    const { t, } = useTranslation();

    const filter = new Filter();

    const handleDisplayNameChange = (event : ChangeEvent<HTMLInputElement>) => {
        const displayName = filter.clean(event.target.value);

        if (onChange) onChange({
            ...bundle,
            displayName,
        });
    };

    const handleDescriptionChange = (event : ChangeEvent<HTMLInputElement>) => {
        const description = filter.clean(event.target.value);

        if (onChange) onChange({
            ...bundle,
            description,
        });
    };

    useEffect(() => {
        if (!(bundle?.displayName) && onChange) onChange({
            ...bundle,
            displayName : t('bundle.create.default_display_name', {
                name : capitalizeFirstLetter(uniqueNamesGenerator({
                    dictionaries : [
                        adjectives,
                        animals,
                        colors,
                        names,
                    ],
                    separator    : ' ',
                    length       : 2,
                })),
            }),
        });
    }, []);

    return (
        <Stack spacing={2}>
            <TextField
                required
                error={(bundle?.displayName ?? '').trim().length === 0}
                disabled={readOnly}
                label={t('bundle.create.display_name')}
                value={bundle?.displayName}
                onChange={handleDisplayNameChange} />
            <TextField
                disabled={readOnly}
                label={t('bundle.create.description')}
                value={bundle?.description}
                onChange={handleDescriptionChange} />
            <Divider />
            <Typography variant='h6'>{t('bundle.create.items')}</Typography>
            <ListView packages={bundle?.packages ?? []} />
        </Stack>
    );
};
