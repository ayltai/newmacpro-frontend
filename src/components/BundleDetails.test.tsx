import React from 'react';

import { POPULAR_APPS, } from '../constants';
import type { Bundle, } from '../types';
import { fireEvent, render, } from '../utils/test';
import { BundleDetails, } from './BundleDetails';

const BUNDLE : Bundle = {
    id          : '1',
    displayName : 'Test Bundle',
    packages    : [
        POPULAR_APPS[0],
    ],
};

describe('<BundleDetails />', () => {
    it('renders correctly', () => {
        expect(render(
            <BundleDetails bundle={BUNDLE} />
        )).toMatchSnapshot();
    });

    it('generates a default display name', () => {
        const { getAllByRole, } = render(
            <BundleDetails bundle={{
                ...BUNDLE,
                displayName : undefined,
            }} />
        );

        expect(getAllByRole('textbox')[0]).toHaveTextContent('');
    });

    it('triggers onChange when the description changes', () => {
        const onChange = jest.fn();

        const { getAllByRole, } = render(
            <BundleDetails
                bundle={BUNDLE}
                onChange={onChange} />
        );

        fireEvent.change(getAllByRole('textbox')[1], {
            target : {
                value : 'Dummy',
            },
        });

        expect(onChange).toBeCalledTimes(1);
        expect(onChange).toBeCalledWith({
            ...BUNDLE,
            description : 'Dummy',
        });
    });
});
