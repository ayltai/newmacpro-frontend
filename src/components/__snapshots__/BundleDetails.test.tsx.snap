// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BundleDetails /> renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiStack-root css-1sazv7p-MuiStack-root"
      >
        <div
          class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-disabled MuiFormLabel-filled Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined css-c5hrqi-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="true"
            for=":r0:"
            id=":r0:-label"
          >
            bundle.create.display_name
            <span
              aria-hidden="true"
              class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
            >
               
              *
            </span>
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary Mui-disabled MuiInputBase-formControl css-1wv3h1b-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input Mui-disabled css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
              disabled=""
              id=":r0:"
              required=""
              type="text"
              value="Test Bundle"
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-14lo706"
              >
                <span>
                  bundle.create.display_name
                   
                  *
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-disabled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-1eqzuua-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
            for=":r1:"
            id=":r1:-label"
          >
            bundle.create.description
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary Mui-disabled MuiInputBase-formControl css-1wv3h1b-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input Mui-disabled css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
              disabled=""
              id=":r1:"
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-yjsfm1"
              >
                <span>
                  bundle.create.description
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <hr
          class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
        />
        <h6
          class="MuiTypography-root MuiTypography-h6 css-1bljjaj-MuiTypography-root"
        >
          bundle.create.items
        </h6>
        <ul
          class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
        >
          <li
            class="MuiListItem-root css-1fymnkd-MuiListItem-root"
          >
            <div
              class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="InventoryIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M20 2H4c-1 0-2 .9-2 2v3.01c0 .72.43 1.34 1 1.69V20c0 1.1 1.1 2 2 2h14c.9 0 2-.9 2-2V8.7c.57-.35 1-.97 1-1.69V4c0-1.1-1-2-2-2zm-5 12H9v-2h6v2zm5-7H4V4l16-.02V7z"
                />
              </svg>
            </div>
            <div
              class="MuiListItemText-root css-tlelie-MuiListItemText-root"
            >
              <span
                class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-15qlsi5-MuiTypography-root"
              >
                Google Chrome
              </span>
            </div>
          </li>
        </ul>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiStack-root css-1sazv7p-MuiStack-root"
    >
      <div
        class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-disabled MuiFormLabel-filled Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined css-c5hrqi-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="true"
          for=":r0:"
          id=":r0:-label"
        >
          bundle.create.display_name
          <span
            aria-hidden="true"
            class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
          >
             
            *
          </span>
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary Mui-disabled MuiInputBase-formControl css-1wv3h1b-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input Mui-disabled css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            disabled=""
            id=":r0:"
            required=""
            type="text"
            value="Test Bundle"
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-14lo706"
            >
              <span>
                bundle.create.display_name
                 
                *
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-disabled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-1eqzuua-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          for=":r1:"
          id=":r1:-label"
        >
          bundle.create.description
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary Mui-disabled MuiInputBase-formControl css-1wv3h1b-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input Mui-disabled css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            disabled=""
            id=":r1:"
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-yjsfm1"
            >
              <span>
                bundle.create.description
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <hr
        class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
      />
      <h6
        class="MuiTypography-root MuiTypography-h6 css-1bljjaj-MuiTypography-root"
      >
        bundle.create.items
      </h6>
      <ul
        class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
      >
        <li
          class="MuiListItem-root css-1fymnkd-MuiListItem-root"
        >
          <div
            class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="InventoryIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M20 2H4c-1 0-2 .9-2 2v3.01c0 .72.43 1.34 1 1.69V20c0 1.1 1.1 2 2 2h14c.9 0 2-.9 2-2V8.7c.57-.35 1-.97 1-1.69V4c0-1.1-1-2-2-2zm-5 12H9v-2h6v2zm5-7H4V4l16-.02V7z"
              />
            </svg>
          </div>
          <div
            class="MuiListItemText-root css-tlelie-MuiListItemText-root"
          >
            <span
              class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-15qlsi5-MuiTypography-root"
            >
              Google Chrome
            </span>
          </div>
        </li>
      </ul>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
